docs/Trident-API-Overview.md

```diff
@@ -299,6 +299,106 @@ Below example shows how these APIs can be used to find maximum using respective
+```
+
+Examples of tumbling and sliding windows can be found [here](Windowing.html)
+
+#### Common windowing API
+Below is the common windowing API which takes `WindowConfig` for any supported windowing configurations. 
+
+```java
+
+    public Stream window(WindowConfig windowConfig, WindowsStoreFactory windowStoreFactory, Fields inputFields,
+                         Aggregator aggregator, Fields functionFields)
+                         
+```
+
+`windowConfig` can be any of the below.
+ - `SlidingCountWindow.of(int windowCount, int slidingCount)`
+ - `SlidingDurationWindow.of(BaseWindowedBolt.Duration windowDuration, BaseWindowedBolt.Duration slidingDuration)`
+ - `TumblingCountWindow.of(int windowLength)`
+ - `TumblingDurationWindow.of(BaseWindowedBolt.Duration windowLength)`
+ 
+ 
+Trident windowing APIs need `WindowsStoreFactory` to store received tuples and aggregated values. Currently, basic implementation 
+for HBase is given with `HBaseWindowsStoreFactory`. It can further be extended to address respective usecases. 
+Example of using `HBaseWindowStoreFactory` for windowing can be seen below.    
+
+```java
+
+    // window-state table should already be created with cf:tuples column
+    HBaseWindowsStoreFactory windowStoreFactory = new HBaseWindowsStoreFactory(new HashMap<String, Object>(), "window-state", "cf".getBytes("UTF-8"), "tuples".getBytes("UTF-8"));
+    FixedBatchSpout spout = new FixedBatchSpout(new Fields("sentence"), 3, new Values("the cow jumped over the moon"),
+            new Values("the man went to the store and bought some candy"), new Values("four score and seven years ago"),
+            new Values("how many apples can you eat"), new Values("to be or not to be the person"));
+    spout.setCycle(true);
+
+    TridentTopology topology = new TridentTopology();
+
+    Stream stream = topology.newStream("spout1", spout).parallelismHint(16).each(new Fields("sentence"),
+            new Split(), new Fields("word"))
+            .window(TumblingCountWindow.of(1000), windowStoreFactory, new Fields("word"), new CountAsAggregator(), new Fields("count"))
+            .peek(new Consumer() {
+                @Override
+                public void accept(TridentTuple input) {
+                    LOG.info("Received tuple: [{}]", input);
+                }
+            });
+
+    StormTopology stormTopology =  topology.build();
+    
+```
+
+Detailed description of all the above APIs in this section can be found [here](javadocs/org/apache/storm/trident/Stream.html)  
+
+#### Example applications
+Example applications of these APIs are located at [TridentHBaseWindowingStoreTopology](https://github.com/apache/storm/blob/master/examples/storm-starter/src/jvm/org/apache/storm/starter/trident/TridentHBaseWindowingStoreTopology.java) 
+and [TridentWindowingInmemoryStoreTopology](https://github.com/apache/storm/blob/master/examples/storm-starter/src/jvm/org/apache/storm/starter/trident/TridentWindowingInmemoryStoreTopology.java) 
```

Comment:
Instead of putting full github links in here, could you please use `{{page.git-blob-base}}` instead.

i.e.

```
Example applications of these APIs are located at [TridentHBaseWindowingStoreTopology]({{page.git-blob-base}}/examples/storm-starter/src/jvm/org/apache/storm/starter/trident/TridentHBaseWindowingStoreTopology.java) 
and [TridentWindowingInmemoryStoreTopology]({{page.git-blob-base}}/examples/storm-starter/src/jvm/org/apache/storm/starter/trident/TridentWindowingInmemoryStoreTopology.java) 
```

That way it will point to the correct thing on github associated with the release this documentation is a part of.